librabbitmq/amqp_socket.h

```diff
@@ -177,9 +177,123 @@ int amqp_open_socket_inner(char const *hostname, int portnumber,
+ * \param [in] timeout a timeout to wait for a RPC answer. Passing in
+ *             NULL will result in blocking behavior.
+ * \return a amqp_rpc_reply_t:
+ *  - r.reply_type == AMQP_RESPONSE_NORMAL. RPC completed successfully
+ *  - r.reply_type == AMQP_STATUS_TIMEOUT the timeout was reached while waiting
+ *    for RPC answer.
+ *  - r.reply_type == AMQP_RESPONSE_SERVER_EXCEPTION. The broker returned an
+ *    exception:
+ *    - If r.reply.id == AMQP_CHANNEL_CLOSE_METHOD a channel exception
+ *      occurred, cast r.reply.decoded to amqp_channel_close_t* to see details
+ *      of the exception. The client should amqp_send_method() a
+ *      amqp_channel_close_ok_t. The channel must be re-opened before it
+ *      can be used again. Any resources associated with the channel
+ *      (auto-delete exchanges, auto-delete queues, consumers) are invalid
+ *      and must be recreated before attempting to use them again.
+ *    - If r.reply.id == AMQP_CONNECTION_CLOSE_METHOD a connection exception
+ *      occurred, cast r.reply.decoded to amqp_connection_close_t* to see
+ *      details of the exception. The client amqp_send_method() a
+ *      amqp_connection_close_ok_t and disconnect from the broker.
+ *  - r.reply_type == AMQP_RESPONSE_LIBRARY_EXCEPTION. An exception occurred
+ *    within the library. Examine r.library_error and compare it against
+ *    amqp_status_enum values to determine the error.
+ */
+amqp_rpc_reply_t
+AMQP_CALL amqp_simple_rpc_noblock(amqp_connection_state_t state,
+                                  amqp_channel_t channel,
+                                  amqp_method_number_t request_id,
+                                  amqp_method_number_t *expected_reply_ids,
+                                  void *decoded_request_method,
+                                  struct timeval *timeout);
+
+
+/**
+ * Sends a method to the broker and waits for a method response
+ *
+ * \param [in] state the connection object
+ * \param [in] channel the channel object
+ * \param [in] request_id the method number of the request
+ * \param [in] reply_id the method number expected in response
+ * \param [in] decoded_request_method the request method
+ * \param [in] timeout a timeout to wait for a RPC answer. Passing in
+ *             NULL will result in blocking behavior.
+ * \return a pointer to the method returned from the broker, or NULL on error.
+ *  On error amqp_get_rpc_reply() will return an amqp_rpc_reply_t with
+ *  details on the error that occurred.
+ */
+void *
+amqp_simple_rpc_decoded_noblock(amqp_connection_state_t state,
+                                amqp_channel_t channel,
+                                amqp_method_number_t request_id,
+                                amqp_method_number_t reply_id,
+                                void *decoded_request_method,
+                                struct timeval *timeout);
+
+int amqp_send_method_inner_noblock(amqp_connection_state_t state,
```

Comment:
For internal functions, don't bother creating a separate `noblock` version, just change the existing method.